### PR TITLE
Moved timestamps and sequence number from descendants to RawBuffer

### PR DESCRIPTION
**Pull request by asahtik**
Getters and setters for moved parameters in depthai-shared and changed (de)serialization. Depends on [depthai-shared](https://github.com/luxonis/depthai-shared/pull/147) MR.

### CODE REVIEW

File: src/pipeline/datatype/Buffer.cpp [hunk 1]
```diff
@@ -22,4 +22,39 @@ void Buffer::setData(std::vector<std::uint8_t>&& data) {
...
+    raw->ts.sec = duration_cast<seconds>(ts).count();
+    raw->ts.nsec = duration_cast<nanoseconds>(ts).count() % 1000000000;
+    return *this;
+}
+Buffer& Buffer::setTimestampDevice(std::chrono::time_point<std::chrono::steady_clock, std::chrono::steady_clock::duration> tp) {
+    // Set timestamp from timepoint
+    using namespace std::chrono;
+    auto ts = tp.time_since_epoch();
+    raw->tsDevice.sec = duration_cast<seconds>(ts).count();
+    raw->ts.nsec = duration_cast<nanoseconds>(ts).count() % 1000000000;
```
Comment (szabi-luxonis) [inline src/pipeline/datatype/Buffer.cpp:52]: should be `tsDevice`

**Review by szabi-luxonis — CHANGES_REQUESTED** [review]

**Review by szabi-luxonis — APPROVED** [review]
LGTM from my side

TODO: port generic sync node from rvc3

**Review by themarpe — CHANGES_REQUESTED** [review]
Looks like the rest of the messages still have `setTimestamp, etc...` functions.

For setters we'd have to CRTP (due to return type), but atleast getters can be removed cleanly
In derived setters, lets call parent & static_cast down cast then aka:
```
NNData& NNData::setTimestamp(std::chrono::time_point<std::chrono::steady_clock, std::chrono::steady_clock::duration> timestamp) {
    static_cast<NNData&>(Buffer::setTimestamp(timestamp));
}
```

**Review by themarpe — CHANGES_REQUESTED** [review]

**Review by themarpe — APPROVED** [review]
Update `depthai-core\cmake\Depthai\DepthaiDeviceSideConfig.cmake` FW version with updated FW that has this changes in as well

**Pull request merged**